src/coreclr/inc/profilepriv.h

```diff
@@ -289,22 +275,25 @@ class ProfControlBlock
     BOOL IsMainProfiler(ProfToEEInterfaceImpl *pProfToEE);
     ProfilerInfo *GetProfilerInfo(ProfToEEInterfaceImpl *pProfToEE);
 
+    template<typename ConditionFunc, typename CallbackFunc, typename Data = void, typename... Args>
+    static void DoProfilerCallbackHelper(ProfilerInfo *pProfilerInfo, ConditionFunc condition, Data *additionalData, CallbackFunc callback, HRESULT *pHR, Args... args)
```

Comment:
I do not think you need the `additionalData` argument. It is NULL in most cases. The few cases where it is not NULL can pass it as an extra argument in `args`.